mpu6050_user/mpu6050.c

```diff
@@ -0,0 +1,148 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/ioctl.h>
+#include <linux/i2c-dev.h>
+#include <linux/i2c.h>
+#include <sys/time.h>
+#include <time.h>
+
+/* MPU6050 internal registers */
+#define WHO_AM_I	0x75
+#define TEMP_OUT_H	0x41
+#define TEMP_OUT_L	0x42
+#define PWR_MGMT_1	0x6b
+
+#define GYRO_ADDR	0x68
+
+//------------------------------------------------------------------------------
+int readData(int fd, u_int8_t regAddr, u_int8_t *buf, int len)
+{
+	// Write memory address again
+	write(fd, &regAddr, 1);
+
+	// Read bytes
+	if (!read(fd, buf, len)) {
+		printf("Failed to read buffer\n");
+		return -1;
+	}
+
+	//printf("Reg addr: 0x%02X, Read byte: 0x%02X\n", regAddr, buf);
+	return 0;
+}
+//------------------------------------------------------------------------------
+int writeData(int fd, u_int8_t regAddr, u_int8_t val)
+{
+	int w = 0;
+	u_int8_t buf[2] = {0};
+
+	buf[0] = regAddr;
+	buf[1] = val;
+
+// Write memory address again
+	w = write(fd, buf, 2);
+	if (w != 2) {
+		printf("Failed to write buffer, err: %d\n", w);
+		return -1;
+	}
+
+	//printf("Reg addr: 0x%02X, Write byte: 0x%02X\n", regAddr, val);
+	return 0;
+}
+
+//------------------------------------------------------------------------------
+int gyro_temp(int i2c_fd, float *temp)
+{
+	u_int8_t buf[2] = {0};
+
+	if (readData(i2c_fd, TEMP_OUT_H, buf, 2) < 0)
+		return -1;
+
+	//printf("Read bytes: 0x%04X\n", buf[1] | (buf[0] << 8));
+	//printf("Val: %d\n", (short)(buf[1] | (buf[0] << 8)));
+
+	*temp = (short)(buf[1] | (buf[0] << 8));
+	*temp /= 340 + 30;
+	*temp += 30;
```

Comment:
Please check the correctness of followed lines:
>int gyro_temp(int i2c_fd, float *temp)
>{
>	u_int8_t buf[2] = {0};
>
>	if (readData(i2c_fd, TEMP_OUT_H, buf, 2) < 0)
>		return -1;
>
>	//printf("Read bytes: 0x%04X\n", buf[1] | (buf[0] << 8));
>	//printf("Val: %d\n", (short)(buf[1] | (buf[0] << 8)));
>
>	*temp = (short)(buf[1] | (buf[0] << 8));
>	*temp /= 340 + 30;
>	*temp += 30;

Reply:
Done.